specification/protocol/otlp.md

```diff
@@ -397,14 +432,41 @@ header.
 
 ##### Success
 
-On success the server MUST respond with `HTTP 200 OK`. Response body MUST be
-Protobuf-encoded `ExportTraceServiceResponse` message for traces,
-`ExportMetricsServiceResponse` message for metrics and
-`ExportLogsServiceResponse` message for logs.
+The success response indicates telemetry data is successfully processed by the
+server. If the server receives an empty request (a request that does not carry
+any telemetry data) the server SHOULD respond with success.
+
+On success, the server MUST respond with `HTTP 200 OK`. The response body MUST be
+a Protobuf-encoded [Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto)
+message (`ExportTraceServiceResponse` for traces,
+`ExportMetricsServiceResponse` for metrics and
+`ExportLogsServiceResponse` for logs).
 
 The server SHOULD respond with success no sooner than after successfully
 decoding and validating the request.
 
+The server MUST leave the `partial_success` field unset
+in case of a successfull response.
+
+##### Partial Success
+
+If the processing of the request is partially successfull
+(i.e. when the server accepts only parts of the data and rejects the rest) the
+server MUST respond with the same message types and HTTP status code as
+specified on the Success case.
+
+Additionally, the server MUST initialize the `partial_success` field
+(`ExportTracePartialSuccess` message for traces,
+`ExportMetricsPartialSuccess` message for metrics and
+`ExportLogsPartialSuccess` message for logs), and it MUST set the respective
+`accepted_spans`, `accepted_data_points` or `accepted_log_records` field with
+the number of spans/data points/log records it accepted.
+
+The server MAY populate the `error_message` field with a human-readable
+error message in English. The protocol does not attempt to define the content
+nor structure of such error message, but it SHOULD generally offer guidance
+on how users can address the issues.
+
```

Comment:
Should we add guidance about when the retry field is set? do we suggest retrying partially accepted things? is this even possible? same for failures i guess, but its easier there...

Reply:
From the proto PRs I think the general agreement is no retry for now in a partial success case. We might be able to, but for that we probably need the extra structure and define semantic conventions on _what_ error is retryable. I will try to add something indicating this so it's clear.

Reply:
That's exactly what I was thinking of. Thanks!